docs/platforms/apple/guides/ios/session-replay/customredact.mdx

```diff
@@ -0,0 +1,66 @@
+---
+title: Using custom redact for Session Replay
+sidebar_order: 5501
+notSupported:
+description: "Learn how to choose which part of your app's data to redact in Session Replay."
+---
+
+<Note>
+
+Using custom redaction for Session Replay can expose sensitive data. Make sure to double-check every part of your app's data that you choose to redact or not.
+
+</Note>
+
+By default, our Session Replay SDK masks all text content, images, and user input, giving you heightened confidence that no sensitive data will leave the device. However, you can choose which parts of your app's data to redact or not by using some different options.
+
+
+## Redact by View class
+
+You can choose which type of view you want to redact or ignore by using the `redactViewClasses` or `ignoreViewClasses` options.
+
+Let's say you have a custom view that you want to redact and a UILabel subclass (which normally would be redacted) that you want to ignore. You can set the options like this:
+
+```swift
+  options.experimental.sessionReplay.redactViewClasses = [MyCustomView.self]
+  options.experimental.sessionReplay.ignoreViewClasses = [MyCustomLabel.self]
+```
+
+## Redact View by instance
+
+You can also choose to redact or ignore a specific view instance by using the replay API or view extensions like this:
+
+```swift
+  SentrySDK.replay.redactView(view: view)
+  SentrySDK.replay.ignoreView(view: label)
+```
+or
+
+```swift
+  view.sentryReplayRedact()
+  label.sentryReplayIgnore()
+```
+
+## SwiftUI
+
+Because of the way SwiftUI is transformed into UIKit, it will often be over-redacted. A modifier like `background` uses the same element as an `Image`.
+In order to control the SwiftUI redact process, you first need to disable the default redaction options:
+
+```swift
+    options.experimental.sessionReplay.redactAllText = false
+    options.experimental.sessionReplay.redactAllImages = false
+```
+
+Then you can manually choose which `View` you want to redact with the `replayRedact` modifier
+
+```swift    
+    var body: some View {
+        Text("Hello, World!")
+            .sentryReplayRedact()
```

Comment:
`m`: Can't you also wrap this with something like as you can for `SentryTracedView`?
```
SentryReplayRedact {
    Text("Hello world")

}
```

If not, maybe we should add it for consistency.

Reply:
No, modifiers don’t work like that. Technically, it would be possible to create a view that wraps another view, but it would be a hack. My understanding is that the modifier approach is the preferred way in SwiftUI.